discovery/consul/consul.go

```diff
@@ -92,6 +93,26 @@ type Discovery struct {
 
 // NewDiscovery returns a new Discovery for the given config.
 func NewDiscovery(conf *config.ConsulSDConfig) (*Discovery, error) {
+	var wrapper *http.Client
+	if &conf.TLSConfig != nil {
+		tls := &consul.TLSConfig{
+			Address:conf.Server,
+			CAFile:conf.TLSConfig.CAFile,
+			CertFile:conf.TLSConfig.CertFile,
+			KeyFile:conf.TLSConfig.KeyFile,
+			InsecureSkipVerify:conf.TLSConfig.InsecureSkipVerify,
+		}
+		consultls, err := consul.SetupTLSConfig(tls)
```

Comment:
This is duplicating the logic we already have, let;s just do this ourselves as we do elsewhere.